public/app/features/alerting/unified/components/rule-editor/QueryOptions.tsx

```diff
@@ -0,0 +1,112 @@
+import { css } from '@emotion/css';
+import React, { useState } from 'react';
+
+import { dateTime, getDefaultRelativeTimeRange, GrafanaTheme2, RelativeTimeRange } from '@grafana/data';
+import { relativeToTimeRange } from '@grafana/data/src/datetime/rangeutil';
+import { clearButtonStyles, Icon, RelativeTimeRangePicker, Toggletip, useStyles2 } from '@grafana/ui';
+import { AlertQuery } from 'app/types/unified-alerting-dto';
+
+import { AlertQueryOptions, DEFAULT_MAX_DATA_POINTS, MaxDataPointsOption } from './QueryWrapper';
+
+export interface QueryOptionsProps {
+  query: AlertQuery;
+  queryOptions: AlertQueryOptions;
+  onChangeTimeRange?: (timeRange: RelativeTimeRange, index: number) => void;
+  onChangeQueryOptions: (options: AlertQueryOptions, index: number) => void;
+  index: number;
+}
+
+export const QueryOptions = ({
+  query,
+  queryOptions,
+  onChangeTimeRange,
+  onChangeQueryOptions,
+  index,
+}: QueryOptionsProps) => {
+  const styles = useStyles2(getStyles);
+
+  const [showOptions, setShowOptions] = useState(false);
+
+  const timeRange = query.relativeTimeRange ? relativeToTimeRange(query.relativeTimeRange) : undefined;
+
+  return (
+    <>
+      <Toggletip
+        content={
+          <div className={styles.container}>
+            <div>
+              {onChangeTimeRange && (
+                <div className={styles.timeRangeContainer}>
+                  <span className={styles.timeRangeLabel}>Time Range</span>
+                  <RelativeTimeRangePicker
+                    timeRange={query.relativeTimeRange ?? getDefaultRelativeTimeRange()}
+                    onChange={(range) => onChangeTimeRange(range, index)}
+                  />
+                </div>
+              )}
+            </div>
+            <div className={styles.queryOptions}>
+              <MaxDataPointsOption
+                options={queryOptions}
+                onChange={(options) => onChangeQueryOptions(options, index)}
+              />
+            </div>
+          </div>
+        }
+        closeButton={true}
+        placement="bottom-start"
+      >
+        <button type="button" className={styles.actionLink} onClick={() => setShowOptions(!showOptions)}>
+          Options {showOptions ? <Icon name="angle-down" /> : <Icon name="angle-right" />}
```

Comment:
When looking at the panel's implementation the arrows are reversed – not a blocker since we can change it later :)

<img width="153" alt="Screenshot 2023-07-04 at 11 02 01" src="https://github.com/grafana/grafana/assets/868844/a7c24228-9664-44ca-8b18-a320d24991b1">